test/test_optimizer.py

```diff
@@ -0,0 +1,85 @@
+import unittest
+import torch
+import torch.utils.optimizer
+from torch.nn import functional as F
+
+batch_size = 2
+input_channels_per_group = 6
+height = 16
+width = 16
+output_channels_per_group = 6
+groups = 4
+kernel_h = kernel_w = 3
+stride_h = stride_w = 1
+pad_h = pad_w = 1
+dilation = 1
+input_channels = input_channels_per_group * groups
+output_channels = output_channels_per_group * groups
+strides = (stride_h, stride_w)
+paddings = (pad_h, pad_w)
+dilations = (dilation, dilation)
+conv_weight_shape = (output_channels, input_channels_per_group, kernel_h, kernel_w)
+conv_bias_shape = (output_channels)
+
+input_data = torch.rand((batch_size, input_channels, height, width))
+conv_weight = torch.rand((output_channels, input_channels_per_group, kernel_h, kernel_w))
+conv_bias = torch.rand((output_channels))
+result = F.conv2d(input_data, conv_weight, conv_bias, strides, paddings, dilations, groups)
+weight_output_dim = 24
+linear_input_shape = result.shape[1]
+linear_weight_shape = (weight_output_dim, linear_input_shape)
```

Comment:
Can you move all of this in `test_optimize_for_mobile` function? We shouldn't have any global variable initialization that is not ncessary.